### PR TITLE
OSDOCS-10345: Adding specific firewall prereqs for ROSA HCP

### PR DESCRIPTION
**Pull request by michaelryanmcneill**
Version(s): enterprise-4.15+

Issue: [OSDOCS-10345](https://issues.redhat.com/browse/OSDOCS-10345)

Link to docs preview:
_Primary changes_:
- https://75118--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_planning/rosa-sts-aws-prereqs.html

_Secondary changes_:
- https://75118--ocpdocs-pr.netlify.app/openshift-dedicated/latest/osd_planning/aws-ccs.html
- https://75118--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa_getting_started_iam/rosa-aws-prereqs.html

QE review:
- [ ] QE has approved this change.

Additional information:

Includes:
- new HCP specific firewall prereqs module
- update to multi-level on the rosa-sts-aws-prereqs page to separate Classic vs. HCP
- update to HCP specific prereq page to use HCP specific module
- change to ROSA Classic/OSD firewall prereqs to require a S3 Gateway VPCE to prevent day one issues with the image registry and to bring the guidelines in line with the proxy prereqs

### CODE REVIEW

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 1]
```diff
@@ -3,27 +3,33 @@
...
 ifeval::["{context}" == "rosa-sts-aws-prereqs"]
 :fedramp:
 endif::[]
-ifeval::["{context}" == "rosa-hcp-aws-prereqs"]
-:fedramp:
-endif::[]
 
 :_mod-docs-content-type: PROCEDURE
+ifeval::["{context}" == "rosa-sts-aws-prereqs"]
+[id="rosa-classic-firewall-prerequisites_{context}"]
```
Comment (EricPonvelle) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:13]: I could be wrong don't think this is working correctly. 

image

I believe that it should be `ifdef::openshift-rosa[]`. If we want to create the conditional for this assembly, we need to add an ifeval at the top and bottom of the module, with some term in the middle like `:rosa-classic:`

```
ifeval::["{context}" == "rosa-sts-aws-prereqs"]
:rosa-classic:
endif::[]
```

Check out this module `/modules/rosa-sts-overview-of-the-default-cluster-specifications.adoc` for some examples

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 2]
```diff
@@ -3,27 +3,33 @@
...
 
-ifdef::openshift-rosa[]
 [IMPORTANT]
 ====
-Only ROSA clusters deployed with PrivateLink can use a firewall to control egress traffic.
+Only clusters deployed with PrivateLink can use a firewall to control egress traffic.
 ====
-endif::[]
 
+ifeval::["{context}" != "rosa-sts-aws-prereqs"]
```
Comment (EricPonvelle) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:26]: Same conditional issue as previously.

**Review by EricPonvelle — COMMENTED** [review]
We need to fix the conditionals, but everything else looks fine otherwise.

**Review by EricPonvelle — COMMENTED** [review]
/lgtm

**Comment by michaelryanmcneill** [comment]
@EricPonvelle comments added as requested. Can you re-review?

File: modules/rosa-hcp-firewall-prerequisites.adoc [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+ifeval::["{context}" == "rosa-sts-aws-prereqs"]
+= ROSA with HCP
+endif::[]
+ifeval::["{context}" != "rosa-sts-aws-prereqs"]
+= AWS firewall prerequisites
+
+This section provides the necessary details that enable you to control egress traffic from your {product-title} with Hosted Control Planes cluster. If you are using a firewall to control egress traffic, you must configure your firewall to grant access to the domain and port combinations below. {product-title} with Hosted Control Planes requires this access to provide a fully managed OpenShift service.
+endif::[]
+
+.Requirements
```
Comment (EricPonvelle) [inline modules/rosa-hcp-firewall-prerequisites.adoc:17]: s/.Requirements/.Prerequisites

Per the documentation guidelines these need to be prerequisites.

File: modules/osd-aws-privatelink-firewall-prerequisites.adoc [hunk 4]
```diff
@@ -3,27 +3,34 @@
...
 This section provides the necessary details that enable you to control egress traffic from your {product-title} cluster. If you are using a firewall to control egress traffic, you must configure your firewall to grant access to the domain and port combinations below. {product-title} requires this access to provide a fully managed OpenShift service.
+endif::[]
+
+.Requirements
```
Comment (EricPonvelle) [inline modules/osd-aws-privatelink-firewall-prerequisites.adoc:31]: s/.Requirements/.Prerequisites

Per the documentation guidelines these need to be prerequisites.

**Review by EricPonvelle — COMMENTED** [review]
@michaelryanmcneill - I missed this on my first pass. Just need to change the requirement headers to prerequisite headers.

**Comment by michaelryanmcneill** [comment]
Superseded by #75197

**Pull request closed without merging**